src/libsystemd-network/sd-dhcp-client.c

```diff
@@ -2015,7 +2015,7 @@ int sd_dhcp_client_send_renew(sd_dhcp_client *client) {
         assert_return(client, -EINVAL);
         assert_return(client->fd >= 0, -EINVAL);
 
-        if (IN_SET(client->state, DHCP_STATE_INIT, DHCP_STATE_INIT_REBOOT, DHCP_STATE_STOPPED))
+        if (!client->lease)
```

Comment:
```if (!client->lease && IN_SET(client->state, DHCP_STATE_INIT, DHCP_STATE_INIT_REBOOT, DHCP_STATE_STOPPED)```) ?

Reply:
In those state, the dhcp client does not have lease. So, it is not necessary.

Reply:
yes